---
title: "CoTEs visualization"
author: "Yahan Zhang"
date: "2023-11-21"
#output: html_document
output: rmarkdown::github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Import the necessary libraries

```{r message=FALSE}
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
```

## Read in the files
```{r message=FALSE}
txdb<-TxDb.Hsapiens.UCSC.hg38.knownGene
peak_core <- readPeakFile('D:/TEs_CoTEs_pipline/data/V2-TEs/non_olap/cores1000/542_L1MA1_Merged.bed')
peak_TE <- readPeakFile('D:/TEs_CoTEs_pipline/data/V2-TEs/non_olap/TEs/542_L1MA1_Merged.bed')
#plotAnnoPie(peakAnno)
#peakAnno = as.data.frame(peakAnno)
```

## Anotate the genomic regions
```{r message=FALSE}
peakAnno_TE<-annotatePeak(peak_TE,tssRegion=c(-3000,3000),TxDb=txdb,annoDb="org.Hs.eg.db")
peakAnno_core<-annotatePeak(peak_core,tssRegion=c(-3000,3000),TxDb=txdb,annoDb="org.Hs.eg.db")
```

## Plot the proportions of genomic regions
```{r}
plotAnnoPie(peakAnno_TE)
plotAnnoPie(peakAnno_core)
```

```{r}
peakAnno = as.data.frame(peakAnno_core)
peakAnno
```

## Since the significant chromosome is chrX, curious about the functions of genes evolved
```{r warning=FALSE}
library(gprofiler2)
filtered_df<-subset(peakAnno,peakAnno$seqnames=="chrX")
genes_list<-filtered_df$SYMBOL
gostres<-gost(query=c(genes_list),organism="hsapiens",significant=TRUE,user_threshold = 0.05)
gostplot(gostres, capped = TRUE, interactive = TRUE)
```

